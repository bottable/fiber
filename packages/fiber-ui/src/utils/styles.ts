import {
  compose,
  space,
  color,
  typography,
  height,
  SpaceProps,
  ColorProps,
  TypographyProps
} from 'styled-system'

export const styleComposition = compose(space, height, color, typography)

export type StyleProps = SpaceProps & ColorProps & TypographyProps
