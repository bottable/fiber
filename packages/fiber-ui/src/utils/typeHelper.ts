export function withProperties<A, B>(component: A, properties: B): A & B {
  Object.keys(properties).forEach((key) => {
    // @ts-ignore
    component[key] = properties[key]
  })
  return component as A & B
}
