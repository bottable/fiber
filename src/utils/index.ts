export * from './typeHelper'
export * from './mergeProps'
