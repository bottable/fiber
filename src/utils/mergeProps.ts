import React from 'react'

export type MergeElementProps<
  T extends React.ElementType,
  P extends object = {}
> = Omit<React.ComponentPropsWithRef<T>, keyof P> & P
