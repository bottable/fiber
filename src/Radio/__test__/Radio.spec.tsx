import { Radio } from '..'

import React from 'react'
import { render } from 'test/utils'

describe('Radio', () => {
  it('renders correctly', () => {
    const radio = render(<Radio />)

    expect(radio).toMatchSnapshot()
  })
})
