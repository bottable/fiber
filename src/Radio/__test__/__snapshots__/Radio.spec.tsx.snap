// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio renders correctly 1`] = `
<label
  className="sc-AxjAm YGZFC"
>
  <span
    className="sc-AxirZ cuAbJJ"
  >
    <input
      checked={false}
      className="sc-AxhCb iSKkyU"
      onChange={[Function]}
      type="radio"
    />
    <span
      checked={false}
      className="sc-AxhUy gFqGol"
    />
  </span>
</label>
`;
